src/suites/cts/validation/createPipelineLayout.spec.ts

```diff
@@ -61,39 +61,38 @@ g.test('number of dynamic buffers exceeds the maximum value', async t => {
   )
 );
 
-g.test('dynamic offsets are only allowed on buffers', t => {
-  const { type, visibility } = t.params;
+g.test('visibility and dynamic offsets', t => {
+  const hasDynamicOffset: boolean = t.params.hasDynamicOffset;
+  const type: GPUBindingType = t.params.type;
+  const visibility: GPUShaderStageFlags = t.params.visibility;
   const info = kBindingTypeInfo[type as GPUBindingType];
 
-  const goodDescriptor = {
-    bindings: [{ binding: 0, visibility, type, hasDynamicOffset: false }],
+  const descriptor = {
+    bindings: [{ binding: 0, visibility, type, hasDynamicOffset }],
   };
 
-  t.device.createPipelineLayout({
-    bindGroupLayouts: [t.device.createBindGroupLayout(goodDescriptor)],
-  });
+  let success = true;
+  if (info.type !== 'buffer' && hasDynamicOffset) success = false;
+  if ((visibility & ~info.validStages) !== 0) success = false;
 
-  const badDescriptor = clone(goodDescriptor);
-  badDescriptor.bindings[0].hasDynamicOffset = true;
-
-  const success = info.type === 'buffer';
   t.expectValidationError(() => {
     t.device.createPipelineLayout({
-      bindGroupLayouts: [t.device.createBindGroupLayout(badDescriptor)],
+      bindGroupLayouts: [t.device.createBindGroupLayout(descriptor)],
     });
   }, !success);
 }).params(
   pcombine(
-    poptions('visibility', kShaderStageCombinations), //
-    poptions('type', kBindingTypes)
+    poptions('type', kBindingTypes), //
+    pbool('hasDynamicOffset'),
+    poptions('visibility', kShaderStageCombinations)
   )
 );
 
 g.test('number of bind group layouts exceeds the maximum value', async t => {
-  const { visibility, type } = t.params;
+  const type: GPUBindingType = t.params.type;
 
   const bindGroupLayoutDescriptor: GPUBindGroupLayoutDescriptor = {
-    bindings: [{ binding: 0, visibility, type }],
+    bindings: [{ binding: 0, visibility: C.ShaderStage.Fragment, type }],
```

Comment:
might add a comment that we'll still need to test the read-only storage in VS stage once it's in

Reply:
It seems that most of the binding count limits with all permutations of usage and visibility happens in createBindGroupLayout.spec.ts `max resources per stage/in pipeline layout` and `max resources per stage/in bind group layout`.

So if the purpose of the test is `number of bind group layouts exceeds the maximum value`, then I question whether we need any `bindings:` and can instead just make a pipelineLayout with 4, and then 4 + 1 empty bind group layouts.

Reply:
yes, and in fact I guess it doesn't really make sense to test multiple binding types here. Maybe the best thing is to test multiple values for `visibility` and just one for the binding type.

Reply:
Also, IIRC the change to this test is unrelated to the title of the PR, sorry (rushed this PR out at end of day)